### PR TITLE
fix: `patchChildSlotNodes` & `scopedSlotTextContentFix` not being applied

### PR DESCRIPTION
**Pull request by johnjenkins**
## What is the current behavior?

GitHub Issue Number: Fixes https://github.com/ionic-team/stencil/issues/6054

TLDR ... `customElement.childNodes`, `customElement.children` monkey-patches are never actually switched-on in Stencil atm. `scopedSlotTextContentFix` doesn't fully patch the `textContent`.

...

Contrary to documentation, the `experimentalSlotFixes` or `slotChildNodesFix` flags have no effect on the `childNodes` or `children` accessors. The patches are only applied if the deprecated `needsShadowDomShim` flag is fulfilled. 

This must be a mistake / a hangover from supporting legacy browsers.

Additionally `scopedSlotTextContentFix` alone doesn't fully patch the `host.textContent` accessor. It adds a half-way patch - which I believe to be more confusing than useful. 

To get the full `textContent` patch users much add both `experimentalScopedSlotChanges` AND `scopedSlotTextContentFix` (`experimentalScopedSlotChanges` alone won't add ANY patch). 

## What is the new behavior?

Adding `experimentalSlotFixes` or `slotChildNodesFix` now patches `childNodes`, `children` and `childElementCount` - they only return the nodes / elements that have been slotted - not the items within the component internals. This more closely mimics shadowRoot enabled components.

I have removed the `experimentalScopedSlotChanges` for the `textContent` patch. 
Users need only apply `scopedSlotTextContentFix` or the unified `experimentalSlotFixes`. 

Additionally, I have rolled-in the `experimentalScopedSlotChanges` flag within the unified `experimentalSlotFixes` flag - I believe this makes sense as `experimentalScopedSlotChanges` will eventually be 'on' by default.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

- Added some spec tests for `dom-extras`. 
- Amended wdio tests (e.g. the `text-content-patch` test now only tests for the slotted text and doesn't return internal text)
- I made some flaky wdio tests slightly more reliable on my local machine. Happy to remove these as out-of-scope.

## Other information

Making these changes did open a slight a can of worms because the codebase has never actually dealt with `host.childNodes` being monkey patched; everything assumes it will return a full list of component internals...

- Within non-shadow vdom slot handling 
- Within other monkey-patched methods 

To fix this, using the precedent of `appendChild` (and similar methods) within `dom-extras.ts`, I have added a `__childNodes` accessor to return a component's 'internal' nodes.

### CODE REVIEW

**Comment by christian-bromann** [comment]
> i.e. Is it genuine or just a flaky, one-off thing?

unfortunately yes, I feel like it may be worth transitioning to Vitest but I haven't had the time to look into that nor find out why it is throwing that error.

**Review by christian-bromann — APPROVED** [review]
Awesome stuff 👏
